airbyte-webapp/src/components/connectorBuilder/types.ts

```diff
@@ -115,117 +122,130 @@ export const DEFAULT_BUILDER_STREAM_VALUES: Omit<BuilderStream, "id"> = {
   },
 };
 
-function getInferredInputList(global: BuilderFormValues["global"]): BuilderFormInput[] {
-  if (global.authenticator.type === "ApiKeyAuthenticator") {
-    return [
-      {
-        key: "api_key",
-        required: true,
-        definition: {
-          type: "string",
-          title: "API Key",
-          airbyte_secret: true,
-        },
+const authTypeToKeyToInferredInput: Record<string, Record<string, BuilderFormInput>> = {
+  NoAuth: {},
+  ApiKeyAuthenticator: {
+    api_token: {
```

Comment:
Looks like these property names always need to match the `key` otherwise it won't recognize inferred inputs vs regular inputs.